src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -348,23 +359,14 @@ SHAMapStoreImp::run()
 
         // will delete up to (not including) lastRotated
         if (validatedSeq >= lastRotated + deleteInterval_ &&
-            canDelete_ >= lastRotated - 1)
+            canDelete_ >= lastRotated - 1 && !health())
```

Comment:
Primarily, it's to prevent writing the log message if no actual work is going to get done. Compare to what it was doing before: log the "rotating" message, then do a `health()` check, then `clearPrior()`.

If any change should be made, I'd think it would be to take the first `health()` check out of `clearPrior()`. What do you think?

Reply:
Yes, removing the first health() from clearPrior() makes sense.

Reply:
Done